From 129e76e0ea923b319555f37ea601dfb974a06bfe Mon Sep 17 00:00:00 2001
From: Tiger Yu <tfyu@codeaurora.org>
Date: Tue, 23 Jan 2018 10:58:33 +0800
Subject: qcacld-2.0: Fix potential buffer overwrite in the 
 htt_t2h_lp_msg_handler

Check for the validity of tx_desc_id when received the htt message of
HTT_T2H_MSG_TYPE_MGMT_TX_COMPL_IND from firmware to ensure the buffer
overwrite does not happen.

Change-Id: I0afc781b7fff303525352b817e7eb60b8b05e4d3
CRs-Fixed: 2157917
---
 CORE/CLD_TXRX/TXRX/ol_tx_send.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CORE/CLD_TXRX/TXRX/ol_tx_send.c b/CORE/CLD_TXRX/TXRX/ol_tx_send.c
index f6b187e..196b623 100644
--- a/CORE/CLD_TXRX/TXRX/ol_tx_send.c
+++ b/CORE/CLD_TXRX/TXRX/ol_tx_send.c
@@ -755,6 +755,9 @@ ol_tx_single_completion_handler(
     struct ol_tx_desc_list_elem_t *td_array = pdev->tx_desc.array;
     adf_nbuf_t  netbuf;
 
+    if (tx_desc_id >= pdev->tx_desc.pool_size)
+        return;
+
     tx_desc = td_array[tx_desc_id].tx_desc;
     tx_desc->status = status;
     netbuf = tx_desc->netbuf;
-- 
cgit v1.1

